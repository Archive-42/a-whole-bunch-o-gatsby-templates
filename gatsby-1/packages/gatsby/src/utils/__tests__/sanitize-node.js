const sanitizeNode = require(`../sanitize-node`)

describe(`node sanitization`, () => {
  let testNode

  beforeEach(() => {
    const circularReference = {}
    circularReference.self = circularReference
    const indirectCircular = {
      down1: {
        down2: {},
      },
    }
    indirectCircular.down1.down2.deepCircular = indirectCircular

    testNode = {
      id: `id1`,
      parent: null,
      children: [],
      unsupported: () => {},
      inlineObject: {
        field: `fieldOfFirstNode`,
        re: /re/,
      },
      repeat: `foo`,
      repeat2: `bar`,
      repeat3: {
        repeat3: {
          test: () => {},
          repeat: `bar`,
        },
      },
      inlineArray: [1, 2, 3, Symbol(`test`)],
      internal: {
        type: `Test`,
        contentDigest: `digest1`,
        owner: `test`,
      },
      circularReference,
      deep: {
        indirectCircular,
      },
    }
  })

  it(`Remove not supported fields / values`, () => {
    const result = sanitizeNode(testNode)
    expect(result).toMatchSnapshot()
    expect(result.unsupported).not.toBeDefined()
    expect(result.inlineObject.re).not.toBeDefined()
    expect(result.inlineArray[3]).not.toBeDefined()
  })

  it(`Doesn't mutate original`, () => {
    sanitizeNode(testNode)
    expect(testNode.unsupported).toBeDefined()
    expect(testNode.inlineObject.re).toBeDefined()
    expect(testNode.inlineArray[3]).toBeDefined()
  })

  it(`Create copy of node if it has to remove anything`, () => {
    const result = sanitizeNode(testNode)
    expect(result).not.toBe(testNode)
  })

  it(`Doesn't create clones if it doesn't have to`, () => {
    const testNodeWithoutUnserializableData = {
      id: `id1`,
      parent: null,
      children: [],
      inlineObject: {
        field: `fieldOfFirstNode`,
      },
      inlineArray: [1, 2, 3],
      internal: {
        type: `Test`,
        contentDigest: `digest1`,
        owner: `test`,
      },
    }

    const result = sanitizeNode(testNodeWithoutUnserializableData)
    // should be same instance
    expect(result).toBe(testNodeWithoutUnserializableData)
  })
})
