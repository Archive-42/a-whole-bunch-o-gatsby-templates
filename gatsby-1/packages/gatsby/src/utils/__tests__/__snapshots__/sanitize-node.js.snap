// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`node sanitization Remove not supported fields / values 1`] = `
Object {
  "children": Array [],
  "circularReference": Object {
    "self": [Circular],
  },
  "deep": Object {
    "indirectCircular": Object {
      "down1": Object {
        "down2": Object {
          "deepCircular": [Circular],
        },
      },
    },
  },
  "id": "id1",
  "inlineArray": Array [
    1,
    2,
    3,
  ],
  "inlineObject": Object {
    "field": "fieldOfFirstNode",
  },
  "internal": Object {
    "contentDigest": "digest1",
    "owner": "test",
    "type": "Test",
  },
  "parent": null,
  "repeat": "foo",
  "repeat2": "bar",
  "repeat3": Object {
    "repeat3": Object {
      "repeat": "bar",
    },
  },
}
`;
